src/legacy/core_plugins/dashboard_embeddable_container/public/np_ready/public/lib/types.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IconType } from '@elastic/eui';
+import {
+  SavedObject as SavedObjectType,
+  SavedObjectAttributes,
+} from '../../../../../../../core/server';
+
+export interface DashboardCapabilities {
+  showWriteControls: boolean;
+  createNew: boolean;
+}
+
+export interface RefreshInterval {
+  pause: boolean;
+  value: number;
+}
+
+export interface TimeRange {
+  from: string;
+  to: string;
+}
+
+// TODO: Replace Saved object interfaces by the once Core will provide when it is ready.
+export type SavedObjectAttribute =
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
+  | SavedObjectAttributes
+  | SavedObjectAttributes[];
+
+export interface SimpleSavedObject<T extends SavedObjectAttributes> {
+  attributes: T;
+  _version?: SavedObjectType<T>['version'];
+  id: SavedObjectType<T>['id'];
+  type: SavedObjectType<T>['type'];
+  migrationVersion: SavedObjectType<T>['migrationVersion'];
+  error: SavedObjectType<T>['error'];
+  references: SavedObjectType<T>['references'];
+  get(key: string): any;
+  set(key: string, value: any): T;
+  has(key: string): boolean;
+  save(): Promise<SimpleSavedObject<T>>;
+  delete(): void;
+}
+
+export interface SavedObjectMetaData<T extends SavedObjectAttributes> {
+  type: string;
+  name: string;
+  getIconForSavedObject(savedObject: SimpleSavedObject<T>): IconType;
+  getTooltipForSavedObject?(savedObject: SimpleSavedObject<T>): string;
+  showSavedObject?(savedObject: SimpleSavedObject<T>): boolean;
+}
+
+// TODO: Replace `IndexPattern` by correct NP import when it is available.
+export interface Field {
+  name: string;
+  type: string;
+  // esTypes might be undefined on old index patterns that have not been refreshed since we added
+  // this prop. It is also undefined on scripted fields.
+  esTypes?: string[];
+  aggregatable: boolean;
+  filterable: boolean;
+  searchable: boolean;
+  parent?: string;
+  subType?: string;
+}
+
+export interface IndexPattern {
```

Comment:
use type  from data/index_patterns

Reply:
same with `TimeRange` and `RefreshInterval`.   Maybe since we aren't moving this to core yet a better interim step would be to import these at the top level and re-export so users in `np_ready` folder can import relatively from up top instead of `ui/*` and we don't have to re-duplicate these types?

Reply:
Yes, for types that are not in NP yet, I'll try to do re-exporting at top level.

Reply:
I removed `IndexPattern` and moved `RefreshInterval` and `TimeRange` to the New Platform.